sql-statements: document ALTER TABLE COMPRESSION='NONE' behavior (#23299)
pingcap/docs: sql-statements/sql-statement-alter-table.md

- `ALTER TABLE t CACHE | NOCACHE` is a TiDB extension to MySQL syntax. For details, see [Cached Tables](/cached-tables.md).

> - `ALTER TABLE ... COMPRESSION` accepts only the string literal `'NONE'` (case-insensitive) and is a no-op. TiDB accepts this statement only for compatibility, for example, for tables migrated from MySQL that carry a `COMPRESSION` attribute. It does not compress or uncompress table data. Other values, such as `'ZLIB'` or `'LZ4'`, return error code 8200 (`This type of ALTER TABLE is currently unsupported`). If you specify an unsupported `COMPRESSION` value together with other options in a single statement, none of the options take effect.

Reviewer: ```suggestion
- `ALTER TABLE ... COMPRESSION` accepts only the string literal `'NONE'` (case-insensitive), but has no effect. TiDB accepts this statement only for MySQL compatibility, for example, for tables migrated from MySQL that carry a `COMPRESSION` attribute. It does not compress or uncompress table data. Other values, such as `'ZLIB'` or `'LZ4'`, return error code 8200 (`This type of ALTER TABLE is currently unsupported`). If you specify an unsupported `COMPRESSION` value together with other options in a single statement, none of the options take effect.
```

Reviewer: reason: "Has no effect" is clearer and more user-friendly than "no-op", while conveying the same behavior.

> - `ALTER TABLE ... COMPRESSION` only accepts `NONE` (case-insensitive) and is a no-op. TiDB accepts this statement only for compatibility (for example, for tables migrated from MySQL that carry a `COMPRESSION` attribute) and does not compress or uncompress any table data. Specifying other values, such as `ZLIB` or `LZ4`, returns the error `This type of ALTER TABLE is currently unsupported` (error code 8200). If you specify an unsupported `COMPRESSION` value together with other options in a single statement, none of the options take effect.

Reviewer: The parser grammar requires `COMPRESSION` to be followed by `stringLit`, so bare `NONE`, `ZLIB`, and `LZ4` do not reflect executable syntax. Keeping the quotes inside the code span also makes the MySQL-migration example immediately usable.

```suggestion
- `ALTER TABLE ... COMPRESSION` accepts only the string literal `'NONE'` (case-insensitive) and is a no-op. TiDB accepts this statement only for compatibility, for example, for tables migrated from MySQL that carry a `COMPRESSION` attribute. It does not compress or uncompress table data. Other values, such as `'ZLIB'` or `'LZ4'`, return error code 8200 (`This type of ALTER TABLE is currently unsupported`). If you specify an unsupported `COMPRESSION` value together with other options in a single statement, none of the options take effect.
```

Author: Applied your suggestion in 375da90 — thanks for catching that the grammar requires a string literal after `COMPRESSION`.
